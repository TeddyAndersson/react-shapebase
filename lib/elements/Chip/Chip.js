import React from 'react';
import _ from 'lodash';
import PropTypes from 'prop-types';
import styled, {css} from 'styled-components';
import SquareShape from '../Shape/SquareShape';
import CircleShape from '../Shape/CircleShape';
import RectangleShape from '../Shape/RectangleShape';
import Paragraph from '../Paragraph/Paragraph';

class Element extends React.Component {

    handleClick = (e) => {
        const { disabled } = this.props

        if (disabled) {
          e.preventDefault()
          return
        }

        _.invoke(this.props, 'onClick', e, this.props)
    }


    render(){
        const {
            children,
            className,
            text,
            ...other
        } = this.props

        let ShapeComponent = RectangleShape;

        return (
            <ShapeComponent className={className}>
                {
                    text ?
                            <Paragraph>
                                {text}
                            </Paragraph>
                        :
                        children
                }
            </ShapeComponent>
        );
    }
}

Element.propTypes = {
  /**
    * Primary content.
    */
  children: PropTypes.node,

  /**
    * Additonal classes.
    */
  className: PropTypes.string,

  /**
     * @ignore
     */
  onClick: PropTypes.func,

  /**
    * Creating a link around the component
    */
  href: PropTypes.string,
}

Element.defaultProps = {
}

const Chip = styled(Element)`
    display: flex;
    justify-content: center;
    align-items: center;
    width: fit-content;
    min-height: 1em;
    padding: 0.4em 0.8em;
    min-width: 1em;
    color: #777;
    background: rgba(255,255,255,0.5);

    ${props => props.scale && css`
        font-size: ${props => props.scale}em;
  `}
`;

export default Chip;
