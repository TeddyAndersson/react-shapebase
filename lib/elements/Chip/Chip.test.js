import React from 'react';
import Chip from './Chip'
import renderer from 'react-test-renderer';

test('should render Chip', () => {
  const component = renderer.create(
    <Chip text="Just a chip"/>
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
})
