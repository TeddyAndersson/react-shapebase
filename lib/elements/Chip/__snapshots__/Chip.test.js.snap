// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Chip 1`] = `
<div
  className="sc-EHOje crPIKJ sc-bxivhb hZGozX sc-bdVaJa fSzclR"
>
  <p
    className="sc-ifAKCX hBvcyL"
  >
    Just a chip
  </p>
</div>
`;
